plaidml/bridge/keras/__init__.py

```diff
@@ -1535,11 +1537,13 @@ def softsign(x):
 
 @_log_call
 def sparse_categorical_crossentropy(target, output, from_logits=False):
-    dims = edsl.TensorDims(output.tensor.shape.ndims)
+    shape = output.tensor.compute_shape()
+    dims = edsl.TensorDims(shape.rank)
```

Comment:
I think this should just be `dims = edsl.TensorDims(output.tensor.rank)`

Reply:
Agreed, avoiding `compute_shape()` is key because its a fairly expensive operation. (which was the motivation for the API change to begin with actually).

Reply:
We can't avoid compute_shape() in this case, because one of the sizes gets passed along in the call one_hot() a couple lines further down; I thought it made more sense to grab the shape object once and reuse it.

Reply:
Is it possible for `one_hot` to take the dim size symbolically?

Reply:
`one_hot` needs to construct a tensor equal to `range(this_size)` where `this_size` is equal to the variable we're talking about. I don't see an immediate way to do this symbolically.

This is in `sparse_categorical_crossentropy` and we haven't yet prioritized handling the sparsity for this function in a performant way at runtime, so I'm not sure bad compile time perf matters that much in this case.